shark/shark/src/main/java/shark/FlatteningFiniteTraversalReferenceReader.kt

```diff
@@ -0,0 +1,70 @@
+@file:Suppress("INVISIBLE_REFERENCE", "INVISIBLE_MEMBER")
+
+package shark
+
+import shark.ChainingInstanceReferenceReader.VirtualInstanceReferenceReader
+import shark.HeapObject.HeapClass
+import shark.HeapObject.HeapInstance
+import shark.HeapObject.HeapObjectArray
+import shark.HeapObject.HeapPrimitiveArray
+import shark.internal.hppc.LongScatterSet
+
+class FlatteningFiniteTraversalReferenceReader(
```

Comment:
Please write a short class description explaining what flattening and finite mean in this context and why we want those things

Reply:
I wrote a description! It is, unfortunately, not very short.